### PR TITLE
Fix !visualprospectingnearspawn

### PR DESCRIPTION
**Pull request by ah-OOG-ah**
This time it should actually work.
TODO:
- [ ] Get other people to reproduce expected behavior (spawn chunks don't scan, command allows them to scan).
- [x] Test on a server.
- ~~Test on maps created before this fix.~~
- [x] Wait for Dream's PR (#29) to be merged.

### CODE REVIEW

**Comment by ah-OOG-ah** [comment]
Removed the automated method, and instead use a command because I can't replicate the bug or the automated fix in a full pack anymore (still can in dev tho). Now /visualprospectingredoserverspawncache still fixes the issue, it's just much faster than the old command for large worlds.

**Comment by ah-OOG-ah** [comment]
yep

File: src/main/java/com/sinthoras/visualprospecting/database/WorldCache.java [hunk 1]
```diff
@@ -77,6 +79,46 @@ public void reset() {
...
+        int spawnChunkX = Utils.coordBlockToChunk(spawn.posX);
+        int spawnChunkZ = Utils.coordBlockToChunk(spawn.posZ);
+
+        // The first corner is 8 chunks less in XZ, and the last is 8 more
+        int startX = spawnChunkX - 8;
+        int startZ = spawnChunkZ - 8;
+        int endX = spawnChunkX + 8;
+        int endZ = spawnChunkZ + 8;
+
+        resetSome(dimID, startX, startZ, endX, endZ);
```
[the remark this answers is not in the document]
Comment (SinTh0r4s) [reply src/main/java/com/sinthoras/visualprospecting/database/WorldCache.java:119]: You can compact the information of your comments into a single variable named `spawnChunksRadius`

**Review by SinTh0r4s — COMMENTED** [review]
Thanks for fixing a bug I caused some time ago! <3

Thought I say thanks by reviewing the code a bit. Haven't touched Minecraft in a long time so I cannot approve it. But I hope you appreciate the feedback. Otherwise just ignore it. I'm glad the project is still maintained!

You are a bit overeager on comments. Comments are immensly helpfull to help the next programmer to understand what you did before, but they can quickly become a burden.

Comments don't age well. The next programmer might add a feature which changes the code, but forgets to update the comment. I don't accuse you of being lazy. Just the others.... and myself ;-)
When another dev wants to use the method he is likely to just read the comment and use it like described which leads to unexpected behaviours and bugs as the comment does not match the method any more.

Or maybe a comment does not add anything but a better name of a method or variable. Where is the value in that? You could just rename the method or variable. That has the added benefit of spreading the comment troughout the code base whereever it is referenced.

On the other side you can document your decisions while writing the code. Like you did on that one loop. That is hands-down just amazing! That helps understanding the code SO MUCH! (because it helps to understand your mindset and enables the next dev to continue your ideas) <3

Go and try out different coding styles, come back after a year or two and see how what helps you best. This is Minecraft modding and not old school MISRA C compliant code from the 90s after all ;-)

Keep up the good work!

File: src/main/java/com/sinthoras/visualprospecting/database/WorldCache.java [hunk 2]
```diff
@@ -77,6 +79,46 @@ public void reset() {
...
+     * @param endZ   The Z coord of the ending chunk.
+     */
+    public void resetSome(int dimID, int startX, int startZ, int endX, int endZ) {
+
+        DimensionCache dim = dimensions.get(dimID);
+        if (dim != null) {
+            dim.clearOreVeins(startX, startZ, endX, endZ);
+        }
+        needsSaving = true;
+        isLoaded = false;
```
[the remark this answers is not in the document]
Comment (ah-OOG-ah) [reply src/main/java/com/sinthoras/visualprospecting/database/WorldCache.java:99]: Nope, this is a copy/paste mistake :laughing:

**Review by eigenraven — APPROVED** [review]

**Pull request merged**